Content.Client/_ES/Wallmount/ESWallmountVisibilitySystem.cs

```diff
@@ -0,0 +1,83 @@
+using System.Numerics;
+using Content.Shared.Throwing;
+using Content.Shared.Wall;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Map;
+using Robust.Shared.Timing;
+
+namespace Content.Client._ES.Wallmount;
+
+/// <summary>
+/// This handles only showing wallmounts which are on "sides of the wall" that your eye can see,
+/// by checking their direction and rotation & comparing to their position relative to your eye.
+/// </summary>
+public sealed class ESWallmountVisibilitySystem : EntitySystem
+{
+    [Dependency] private readonly IEyeManager _eye = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+    [Dependency] private readonly TransformSystem _xform = default!;
+    [Dependency] private readonly SpriteSystem _sprite = default!;
+
+    /// <inheritdoc/>
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        if (!_timing.IsFirstTimePredicted)
+            return;
+
+        if (_player.LocalEntity == null)
+            return;
+
+        var playerXform = Transform(_player.LocalEntity.Value);
+        if (playerXform.MapID != _eye.CurrentEye.Position.MapId)
+            return;
+
+        var playerCoords = _eye.MapToScreen(_xform.GetMapCoordinates(playerXform));
+        if (!playerCoords.IsValid)
+            return;
+
+        var enumerator = AllEntityQuery<WallMountComponent, SpriteComponent, TransformComponent>();
```

Comment:
This is a performance concern in replays. You probably want to set up a component tree for this so you can do accelerated lookups of what is in the viewport bounds. (which should be ~par for perf in normal gameplay, and notably better in replays)

Reply:
yeah, probably should have just used a componenttree from the start, shouldnt be that hard to do